src/renderer/components/app.tsx

```diff
@@ -72,9 +75,58 @@ export class App extends React.Component {
     return workloadsURL();
   }
 
+  getTabLayoutRoutes(page: RegisteredPage) {
+    const menuItem = clusterPageMenuRegistry.getByPage(page);
+    const routes: TabLayoutRoute[] = [];
+    if (!menuItem.id) {
+      return routes;
+    }
+    clusterPageMenuRegistry.getSubItems(menuItem.id).forEach((item) => {
+      const page = clusterPageRegistry.getByPageMenuTarget(item.target);
+      if (page) {
+        routes.push({
+          routePath: page.routePath,
+          url: getExtensionPageUrl({ extensionId: page.extensionId, pageId: page.id, params: item.target.params }),
+          title: item.title,
+          component: page.components.Page,
+          exact: page.exact
+        });
+      }
+    });
+    if (routes.length > 0) {
+      routes.unshift({
+        routePath: page.routePath,
+        url: getExtensionPageUrl({ extensionId: page.extensionId, pageId: page.id, params: menuItem.target.params }),
+        title: <Trans>Overview</Trans>,
+        component: page.components.Page,
+        exact: page.exact
+      })
+    }
```

Comment:
Why do we force an `Overview` page, that seems weird.

Reply:
See the attached screenshot. Extension basically does not know if something registers as a sub-menu... it's better to create consistent UI with "Overview" than duplicate top-level menu name in a first sub-menu item.

Reply:
Wouldn't the API be better then if it does know if it registers as a sub menu? I see two main problems: one, is that there is no way to change that title text; two, there doesn't seem to be a way to remove this, if the extension developer doesn't want it.

Reply:
Example:
- extension A registers a menu "Foo"
- extension B registers a sub-menu "Bar" to "Foo"

Reply:
Okay, so then have a tab layout of size 1.

Reply:
Also, I don't see any handling of the following case:

- extension A registers a menu "Foo"
- extension B registers a sub-menu "Bar" to "Fooo" (by accident). 

should an error be reported (somewhere)?

Reply:
> Okay, so then have a tab layout of size 1.

Not sure if I understand your point. Our menu does not really work like that?

Reply:
But other (builtin) submenus don't have an overview page

What about something like:
```
  clusterPageMenus = [
    {
      target: { submenu: "example" },
      title: "Example Extension",
      components: {
        Icon: ExampleIcon,
      }
    },
    {
      parentId: "example",
      target: { pageId: "example1", params: {} },
      title: "Overview",
      components: {
        Icon: ExampleIcon,
      }
    },
    {
      parentId: "example",
      target: { pageId: "example2", params: {} },
      title: "Sub menu 1",
      components: {
        Icon: ExampleIcon,
      }
    }
  ]
```

Reply:
I agree with Jim, this shouldn't be "automagically" added. And with regards to the tab layout not working like that. It seems very strange that the tab layout should care how many entries it has at all.

Reply:
Ok, based on discussions (on Slack) I'll limit scope of this PR to work only within an extension. We can improve (and introduce more possibilities) on follow up PRs.

Reply:
Sounds good